common/cmake/file_targets.cmake

```diff
@@ -264,8 +264,12 @@ function(ADD_FILE_TARGET)
   endforeach()
 
   if(NOT ${ADD_FILE_TARGET_GENERATED} AND NOT ${ADD_FILE_TARGET_ABSOLUTE})
+    get_filename_component(DEST_PATH ${CMAKE_CURRENT_BINARY_DIR}/${ADD_FILE_TARGET_FILE} DIRECTORY)
     add_custom_command(
       OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/${ADD_FILE_TARGET_FILE}
+      COMMAND
```

Comment:
This probably means you have some weird `add_file_target` that contain paths in the name.  Probably a bad idea (consider something like `../../example.v`).  Probably worth scrubbing the `add_file_target` and removing the `weird` ones.  Also consider generating a warning and/or error on "weird" paths.

Reply:
So you are suggesting that `add_file_target` should only be used for targets that write to the "current binary directory"? I think that we already violate that for all file targets added in `add_fpga_target`.